File: testserver/src/main/java/io/lepo/lukki/testserver/PageHtmlFormatter.java
package io.lepo.lukki.testserver;

import java.util.List;
import java.util.stream.Collectors;

final class PageHtmlFormatter {

  private PageHtmlFormatter() {
  }

  static String format(String pageLink, List<String> links) {
    String linksHtml = formatLinks(links);
    String headerHtml = String.format("<head><title>%s</title></head>", pageLink);
    return String.join("\n",
        "<html>",
        headerHtml,
        "<body><ul>",
        linksHtml,
        "</ul></body>",
        "</html>"
    );
  }

  private static String formatLinks(List<String> links) {
    return links.stream()
        .map(link -> String.format("<li><a href=\"%s\">%s</a></li>", link, link))
        .collect(Collectors.joining("\n", "<ul>", "</ul>"));
  }
}

File: testserver/src/main/kotlin/io/lepo/lukki/testserver/PageHtmlFormatter.kt
package io.lepo.lukki.testserver

import java.util.stream.Collectors

internal object PageHtmlFormatter {

    fun format(pageLink: String, links: List<String>): String {
        val linksHtml = formatLinks(links)
        val headerHtml = String.format("<head><title>%s</title></head>", pageLink)
        return arrayOf(
                "<html>",
                headerHtml,
                "<body><ul>",
                linksHtml,
                "</ul></body>",
                "</html>"
        ).joinToString("\n")
    }

    private fun formatLinks(links: List<String>): String =
            links.stream()
                    .map { String.format("<li><a href=\"%s\">%s</a></li>", it, it) }
                    .collect(Collectors.joining("\n", "<ul>", "</ul>"))
}
